action/protocol/staking/candidate_statemanager.go

```diff
@@ -16,10 +18,17 @@ import (
 )
 
 type (
-	// CandidateStateManager is candidate manager on top of StateMangaer
+	// CandidateBucketCenter contains candidate center and bucket pool
+	CandidateBucketCenter interface {
+		// TODO: remove CandidateCenter interface, return *candCenter
+		CandCenter() CandidateCenter
+		BucketPool() *bucketPool
```

Comment:
Not normal public interface public method return private struct

Reply:
change to private interface